lib/livebook/zta/livebook_teams.ex

```diff
@@ -103,11 +103,11 @@ defmodule Livebook.ZTA.LivebookTeams do
   defp request_user_authentication(conn, team) do
     case Teams.Requests.create_auth_request(team) do
       {:ok, %{"authorize_uri" => authorize_uri}} ->
-        current_url = LivebookWeb.Endpoint.url() <> conn.request_path <> "?teams_identity"
+        redirect_to_url = build_redirect_to_url(conn)
```

Comment:
Try this:

```suggestion
        conn = html(conn, """
        <!DOCTYPE html>
        <html lang="en">
          <head>
            <meta charset="UTF-8">
            <title>Redirecting...</title>
            <script>
              (function() {
                const url = new URL("#{authorize_uri}");
                url.searchParams.set("redirect_to", window.location.href);
                window.location.href = url.toString();
              })();
            </script>
          </head>
        </html>
        """)

        {halt(conn), nil}
```

Reply:
We need to add `teams_identity` param:

```js
const redirectTo = new URL(window.location.href);
redirectTo.searchParams.append("teams_identity", "");
url.searchParams.set("redirect_to", redirectTo.toString());
```

Reply:
Let's also add a comment explaining why we do this :)